Sources/Optimizely/OptimizelyClient.swift

```diff
@@ -591,12 +589,94 @@ open class OptimizelyClient: NSObject {
                                  feature: featureFlag,
                                  featureEnabled: featureEnabled,
                                  variableKey: variableKey,
-                                 variableType: typeName,
+                                 variableType: variable.type,
                                  variableValue: value)
         
         return value
     }
     
+    /// Gets all the variables for a given feature.
+    ///
+    /// - Parameters:
+    ///   - featureKey: The key for the feature flag.
+    ///   - userId: The user ID to be used for bucketing.
+    ///   - attributes: The user's attributes.
+    /// - Returns: all the variables for a given feature.
+    /// - Throws: `OptimizelyError` if feature parameter is not valid
+    public func getAllFeatureVariables(featureKey: String,
+                                       userId: String,
+                                       attributes: OptimizelyAttributes? = nil) throws -> OptimizelyJSON {
+        guard let config = self.config else { throw OptimizelyError.sdkNotReady }
+        var variableMap = [String: Any]()
+        var enabled = false
+        
+        guard let featureFlag = config.getFeatureFlag(key: featureKey) else {
+            throw OptimizelyError.featureKeyInvalid(featureKey)
+        }
+        
+        let decision = self.decisionService.getVariationForFeature(config: config,
+                                                                   featureFlag: featureFlag,
+                                                                   userId: userId,
+                                                                   attributes: attributes ?? OptimizelyAttributes())
+        if let featureEnabled = decision?.variation?.featureEnabled {
+            enabled = featureEnabled
+        }
+        
+        for (_, v) in featureFlag.variablesMap {
+            var featureValue = v.value
+            if enabled, let variable = decision?.variation?.getVariable(id: v.id) {
+                featureValue = variable.value
+            }
+            
+            var valueParsed: Any? = featureValue
+            var isValidType = true
+            
+            if let valueType = Constants.VariableValueType(rawValue: v.type) {
```

Comment:
I like this enum for types. This enum may cause an issue when we see unknown types (forward compatibility). Can we have some tests with "future" variable types?

Reply:
for unknown/future types, the value is still be mapped in the variableMap as string, i have now added an additional test for this.